### PR TITLE
feat: create MR with custom target branch

### LINKED ISSUE #69: Create MR with a custom target branch
Right now, the MR is created with the target branch being the same as the one that's running the CI.

In our case, we use a branch `dev` for development and we'd like to merge all changes into `main` using the MR that changeset created. Problem is as I said, the target branch is always `dev` as it's using `CI_COMMIT_REF_NAME`. Could an option be added to specify a custom target branch? (Similar to how changeset has a `baseBranch` configuration option for comparisons)

As this is a quite simple change, I'd be uo tp opening a PR with such changes, maybe with a new `INPUT_TARGET_BRANCH` env variable?

### PR DESCRIPTION
**Pull request by lluiscab**
This pr introduces a new input variable `INPUT_TARGET_BRANCH`. If set, this value will change the target branch of the created Merge Request. For context, see #69

Resolves #69

### CODE REVIEW

**Comment by lluiscab** [comment]
I've been able to test and confirm that the changes work correctly. PR is ready for revieew

File: src/run.ts [hunk 1]
```diff
@@ -286,27 +289,29 @@ ${
...
-    target_branch: branch,
+    target_branch: mrTargetBranch,
     maxPages: 1,
     perPage: 1,
   })
   console.log(JSON.stringify(searchResult, null, 2))
   if (searchResult.length === 0) {
-    console.log('creating merge request')
+    console.log(
+      `Creating merge request from ${versionBranch} to ${mrTargetBranch}.`,
```
Comment (JounQin) [inline src/run.ts:299]: If the two are same, there is no need to mention it or the message can be simpler.

File: src/run.ts [hunk 2]
```diff
@@ -286,27 +289,29 @@ ${
...
       versionBranch,
-      branch,
+      mrTargetBranch,
       finalMrTitle,
       {
         description: await mrBodyPromise,
       },
     )
   } else {
+    console.log('Found existing merge request, updating.')
```
Comment (JounQin) [inline src/run.ts:311]: Why this change is needed?

File: src/run.ts (hunk 1 above)
Comment (lluiscab) [reply src/run.ts:299]: Assuming running on branch `dev`. Then this message has two variants:

_if there's no `INPUT_TARGET_BRANCH` set_ 

> Creating merge request from changeset-release/dev to dev

_if `INPUT_TARGET_BRANCH` set to `main`_ 


> Creating merge request from changeset-release/dev to main

In both cases, `versionBranch` and `mrTargetBranch` are not the same and it's an useful log to have

File: src/run.ts (hunk 2 above)
Comment (lluiscab) [reply src/run.ts:311]: consistency

Comment (JounQin) [reply src/run.ts:311]: Or align

https://github.com/changesets/action/blob/398d7edfb130ad2d4e076a573724f435ea87cea6/src/run.ts#L359

Besides, can you un-capitalize the messages for consistency with upstream?

Comment (lluiscab) [reply src/run.ts]: done!

File: src/run.ts [hunk 3]
```diff
@@ -286,27 +289,29 @@ ${
...
       versionBranch,
-      branch,
+      mrTargetBranch,
       finalMrTitle,
       {
         description: await mrBodyPromise,
       },
     )
   } else {
+    console.log(`updating found merge request !${searchResult[0].iid}`)
```
Comment (JounQin) [inline src/run.ts:311]:
```suggestion
    console.log(`updating found merge request ${searchResult[0].iid}!`)
```

File: .changeset/seven-pugs-provide.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+'changesets-gitlab': patch
+---
+
+merge requests target branch can now be configured through a input environment variable
```
Comment (JounQin) [inline .changeset/seven-pugs-provide.md:5]:
```suggestion
'changesets-gitlab':  minor
---

feat: merge requests target branch can now be configured through a input environment variable
```

**Review by JounQin — CHANGES_REQUESTED** [review]
@lluiscab Sorry, README.md should be updated at the same time

**Comment by lluiscab** [comment]
> @lluiscab Sorry, README.md should be updated at the same time

Updated!

**Review by JounQin — APPROVED** [review]

**Comment by JounQin** [comment]
@lluiscab Thanks for your contribution!
